### PR TITLE
Fix jsnum's Remainder and Exponentiate funcs

### LINKED ISSUE #2117: jsnum.Remainder and others may not behave correctly
See #515; this may not behave properly.

Noting this to get it out of my inbox.

### PR DESCRIPTION
**Pull request by jakebailey**
Fixes #2117 

Remainder was just wrong. I wrote this code exactly as the spec said, but did not consider rounding.

Exponentiate was only sort of wrong. ECMAScript actually allows this to have implementation-defined behavior. We just used `math.Pow`, but this can technically be off. In some testing, we were wrong for a quarter of the cases. More testing revealed that all engines disagree about many edge cases, coming down to which algorithm they used and even which compiler and compiler options they were compiled with. But, they were all within one "ULP" of each other, which we can at least still do.

Better would be to reject such expressions when stringifying, somewhere, but I don't know where that is.

Also in this PR is testing against Node, like we do with string stuff.

### CODE REVIEW

File: internal/jsnum/jsnum_test.go [hunk 1]
```diff
@@ -26,6 +60,160 @@ func numberToBits(n Number) uint64 {
...
+
+func uint32sToNum(a [2]uint32) Number {
+	bits := uint64(a[0]) | uint64(a[1])<<32
+	return numberFromBits(bits)
+}
+
+func nodeAvailable() bool {
+	_, err := exec.LookPath("node")
+	return err == nil
+}
```
[the remark this answers is not in the document]
Comment (RyanCavanaugh) [reply internal/jsnum/jsnum_test.go]: "Does Corsa's compile-time evaluator correctly handle -0 vs +0", a question no one was asking
